drivers/modem/hl78xx/hl78xx_sockets.c

```diff
@@ -0,0 +1,2378 @@
+		return 0;
+	}
+
+	snprintk(buf, sizeof(buf), "AT+KTCPDEL=%d", sock->id);
+
+	ret = modem_dynamic_cmd_send(socket_data_lcl->mdata_global, NULL, buf, strlen(buf),
+				     allow_matches, 2, false);
+	if (ret < 0) {
+		LOG_ERR("%s ret:%d", buf, ret);
+	}
+
+	return ret;
+}
+
+/*
+ * Socket Offload OPS
+ */
+
+static int offload_socket(int family, int type, int proto)
+{
+	int ret;
+#ifdef CONFIG_MODEM_HL78XX_LOG_CONTEXT_VERBOSE_DEBUG
+	LOG_DBG("%d %d %d %d", __LINE__, family, type, proto);
+#endif
+	/* defer modem's socket create call to bind() */
+	ret = modem_socket_get(&socket_data_global->socket_config, family, type, proto);
+	if (ret < 0) {
+		errno = ret;
+		return -1;
+	}
+
+	errno = 0;
+	return ret;
+}
+
+static int offload_close(void *obj)
+{
+	struct modem_socket *sock = (struct modem_socket *)obj;
+	struct hl78xx_socket_data *socket_data_lcl = hl78xx_socket_data_from_sock(sock);
+
+	if (!sock) {
+		return -EINVAL;
+	}
+	/* Recover the containing instance; guard in case sock isn't from this driver */
+
+	if (!socket_data_lcl) {
+		LOG_ERR("Invalid socket container");
+		return -EINVAL;
+	}
+
+	/* make sure socket is allocated and assigned an id */
+	if (modem_socket_id_is_assigned(&socket_data_lcl->socket_config, sock) == false) {
+		return 0;
+	}
+	if (validate_socket(sock) == 0) {
+		socket_close(socket_data_lcl, sock);
+		socket_delete(socket_data_lcl, sock);
+		modem_socket_put(&socket_data_lcl->socket_config, sock->sock_fd);
+		sock->is_connected = false;
+	}
+	/* Consider here successfully socket is closed */
+	return 0;
+}
+
+static int offload_bind(void *obj, const struct sockaddr *addr, socklen_t addrlen)
+{
+	struct modem_socket *sock = (struct modem_socket *)obj;
+	struct hl78xx_socket_data *socket_data_lcl = hl78xx_socket_data_from_sock(sock);
+
+	/* sanity check: does parent == parent->offload_dev->data ? */
+	if (socket_data_lcl && socket_data_lcl->offload_dev &&
+	    socket_data_lcl->offload_dev->data != socket_data_lcl) {
+		LOG_WRN("parent mismatch: parent != offload_dev->data (%p != %p)", socket_data_lcl,
+			socket_data_lcl->offload_dev->data);
+	}
+#ifdef CONFIG_MODEM_HL78XX_LOG_CONTEXT_VERBOSE_DEBUG
+	LOG_DBG("%d", __LINE__);
+#endif
+	/*  Save bind address information */
+	memcpy(&sock->src, addr, sizeof(*addr));
+	/* Check if socket is allocated */
+	if (modem_socket_is_allocated(&socket_data_lcl->socket_config, sock)) {
+		/* Trigger socket creation */
+		if (create_socket(sock, addr, socket_data_lcl) < 0) {
+			LOG_ERR("%d %s SOCKET CREATION", __LINE__, __func__);
+			return -1;
+		}
+	}
+	return 0;
+}
+
+static int offload_connect(void *obj, const struct sockaddr *addr, socklen_t addrlen)
+{
+	struct modem_socket *sock = (struct modem_socket *)obj;
+	struct hl78xx_socket_data *socket_data_lcl = hl78xx_socket_data_from_sock(sock);
+	int ret = 0;
+	char cmd_buf[sizeof("AT+KTCPCFG=#\r")];
+	char ip_str[NET_IPV6_ADDR_LEN];
+
+#ifdef CONFIG_MODEM_HL78XX_LOG_CONTEXT_VERBOSE_DEBUG
+	LOG_DBG("%d", __LINE__);
+#endif
```

Comment:
I am seeing these all over the code, would it make sense to have a separate macro that hides the ifdef. So something like (not tested)
```
#define DBG(str, ...) IF_ENABLED(CONFIG_MODEM_HL78XX_LOG_CONTEXT_VERBOSE_DEBUG, (LOG_DBG(str, __VA_ARGS__)))
```